querydsl-core/src/main/java/com/querydsl/core/types/TemplateFactory.java

```diff
@@ -223,7 +229,11 @@ public Template create(String template) {
                         }
                         break;
                 }
-                if (asString) {
+                if (operand != null) {
+                    Operator operator = OPERATORS.get(operand.substring(0, 1));
+                    int index2 = Integer.parseInt(operand.substring(1));
```

Comment:
I don't quite like these `substring`s, it's these that I removed in the first place. :/
What's your opinion @johnktims?